### PR TITLE
Add hex int pattern entry to CONTRIBUTING

### LINKED ISSUE #216: Document hex/int pattern
Here's the gist of the pattern (from @ericnordelo):
- using only `int` internally
- convert the input to `int`(from config files, starknet cli, etc..) with the `normalize_number` method
- convert to `hex` before writing to files (like in address for accounts)

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #216

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -58,6 +58,27 @@ As a contributor, you are expected to fork this repository, work on your own for
 
     *IMPORTANT* Please pay attention to the maintainer's feedback, since it's a necessary step to keep up with the standards OpenZeppelin Nile attains to.
 
+
+## Design Patterns
+
+We always try to follow best practices and design patterns to make the code readable and maintainable. While most of them are left to be decided by the PR developer and the reviewer, we require following some patterns we enforce ourselves, presented below:
```
Comment (andrew-fleming) [inline CONTRIBUTING.md:64]:
```suggestion
We always try to follow best practices and design patterns to make the code readable and maintainable. While most of them are left to be decided by the PR developer and the reviewer, we require following some patterns that we enforce ourselves which are presented below:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -58,6 +58,27 @@ As a contributor, you are expected to fork this repository, work on your own for
...
+
+The pattern is defined as follows:
+
+#### Definitions
+- **Internal API** is the set of functions that are not meant to be called directly in scripting, tests, or any other context from users of Nile, but only internally, from other Nile code (including plugins).
+- **Public API** is the complement.
+
+#### Rules
+- Accept and use **only int for Internal API** (for hashes, keys, or addresses).
+- Accept **both hex and int from external input** (config files, starknet cli output, etc..) **or Public API functions**, and convert the input to int before using it internally (use `nile.utils.normalize_number`).
```
Comment (andrew-fleming) [inline CONTRIBUTING.md:76]:
```suggestion
- Accept **both hex and int from external input** (config files, starknet cli output, etc.) **or Public API functions**, and convert the input to int before using it internally (use `nile.utils.normalize_number`).
```

**Review by andrew-fleming — DISMISSED** [review]
Very nice, @ericnordelo! I left a couple minor suggestions, but I think we're looking good :)

**Review by andrew-fleming — APPROVED** [review]
Looks good to me!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -58,6 +58,27 @@ As a contributor, you are expected to fork this repository, work on your own for
...
+## Design Patterns
+
+We always try to follow best practices and design patterns to make the code readable and maintainable. While most of them are left to be decided by the PR developer and the reviewer, we require following some patterns that we enforce ourselves which are presented below:
+
+### Hex/Int pattern
+
+The pattern is defined as follows:
+
+#### Definitions
+- **Internal API** is the set of functions that are not meant to be called directly in scripting, tests, or any other context from users of Nile, but only internally, from other Nile code (including plugins).
```
Comment (martriay) [inline CONTRIBUTING.md:71]:
```suggestion
- **Internal API** is the set of functions that are not meant to be called directly in scripting, tests, or any other context from users of Nile, but only internally from other Nile code (including plugins).
```

**Review by martriay — DISMISSED** [review]

**Review by martriay — APPROVED** [review]

**Pull request merged**